### PR TITLE
opentype.js: update to 0.7.3, add more tests

### PR DESCRIPTION
**Pull request by edzis**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/nodebox/opentype.js/compare/0.7.3...master
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Review by danmarshall — APPROVED** [review]

**Comment by danmarshall** [comment]
Thanks @edzis !

**Comment by edzis** [comment]
@danmarshall CI is breaking because of interoperability with `maker.js` types, where you seem to be the maintainer - `Error: maker.js depends on opentype.js but has a lower required TypeScript version.` Can I or you upgrade the version in that lib?
I added that version because of `Partial`.

**Comment by danmarshall** [comment]
@edzis Maker.js has typings in its own package, so I suppose that it can be deleted altogether from DefinitelyTyped. 

Given that if I delete it, that would be a separate PR which would cause delay to your PR. So I would ask if you can make the upgrade in your PR, if you don't mind.

File: types/opentype.js/index.d.ts [hunk 1]
```diff
@@ -1,65 +1,153 @@
-// Type definitions for opentype.js
+// Type definitions for opentype.js 0.7.3
```
Comment (DanielRosenwasser) [inline types/opentype.js/index.d.ts]: Looks like our lint rules are disabled for this package. Can you enable the lints in our `tslint.json` and fix up the version string to only be a `MAJOR.MINOR` version (e.g. `4.7`), as opposed to including the patch or containing a leading `v` (e.g. `v1.2.3`).

**Review by DanielRosenwasser — CHANGES_REQUESTED** [review]

File: types/opentype.js/index.d.ts (hunk 1 above)
Comment (edzis) [reply types/opentype.js/index.d.ts]: @DanielRosenwasser this is now fixed

**Review by DanielRosenwasser — APPROVED** [review]

**Pull request merged**